executor/show.go

```diff
@@ -1233,8 +1233,22 @@ func appendPartitionInfo(partitionInfo *model.PartitionInfo, buf *bytes.Buffer,
 		return
 	}
 	if partitionInfo.Type == model.PartitionTypeHash {
-		fmt.Fprintf(buf, "\nPARTITION BY HASH( %s )", partitionInfo.Expr)
-		fmt.Fprintf(buf, "\nPARTITIONS %d", partitionInfo.Num)
+		fmt.Fprintf(buf, "\n/*!50100 PARTITION BY HASH (%s)", partitionInfo.Expr)
+		if partitionInfo.Num == 0 {
+			return
+		}
+
+		fmt.Fprintf(buf, "\n(")
+		for i, def := range partitionInfo.Definitions {
+			fmt.Fprintf(buf, "PARTITION %s COMMENT = '%s' ENGINE = InnoDB", def.Name.O, def.Comment)
```

Comment:
I think this is incomplete, it should add all set or non-default table options, not only COMMENT or ENGINE (and also not include them when not set, depending on how strict we want to be compatible with MySQL).

Reply:
the default name defined before here. the[ test case](https://github.com/pingcap/tidb/pull/29931/files#L532) shows the result